homework/index.js

```diff
@@ -30,18 +15,107 @@
+      class: 'contributorHeader',
+    });
+    const contributorWrapper = createAndAppend('div', contributorSection, {
+      class: 'contributorWrapper',
+    });
+    return fetch(data.contributors_url)
+      .then(contributorData => contributorData.json())
+      .then(contData => {
+        contData.forEach(contributor => {
+          const contributorInfo = createAndAppend('a', contributorWrapper, {
+            href: contributor.html_url,
+            target: '_blank',
+            class: 'contributorWrapper',
+          });
+          const contributorDiv = createAndAppend('div', contributorInfo, {
+            class: 'contributor',
+          });
+          createAndAppend('img', contributorDiv, {
+            class: 'contImage',
+            src: contributor.avatar_url,
+          });
+          createAndAppend('h4', contributorDiv, { class: 'contName', text: contributor.login });
+          createAndAppend('span', contributorDiv, {
+            class: 'contNumber',
+            text: contributor.contributions,
+          });
+        });
+      })
+      .catch(err => {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+      });
+  }
+
+  function fetchSingleRepoData(data, index, wrapper) {
+    const filteredData = data.filter(el => el.id === Number(index));
+
+    wrapper.innerHTML = '';
+    const repositoriesSection = createAndAppend('div', wrapper, { class: 'repInfo-side' });
+    const repositoryName = createAndAppend('div', repositoriesSection, { class: 'repName' });
+    const repositoryDescription = createAndAppend('div', repositoriesSection, {
+      class: 'repDescription',
+    });
+    const repositoryForks = createAndAppend('div', repositoriesSection, {
+      class: 'repForks',
+    });
+    const repositoryDate = createAndAppend('div', repositoriesSection, { class: 'repDate' });
+    filteredData.forEach(allData => {
+      createAndAppend('span', repositoryName, { text: 'Repository Name:' });
+      createAndAppend('a', repositoryName, {
+        text: allData.name,
+        href: allData.html_url,
+        target: '_blank',
+      });
+      createAndAppend('span', repositoryDescription, { text: 'Description:' });
+      createAndAppend('p', repositoryDescription, { text: allData.description });
+      createAndAppend('span', repositoryForks, { text: 'Forks:' });
+      createAndAppend('p', repositoryForks, { text: allData.forks });
+      createAndAppend('span', repositoryDate, { text: 'Date:' });
+      createAndAppend('p', repositoryDate, { text: allData.updated_at });
+      /* Fetch Contributor Data */
+      fetchContributorsData(allData, wrapper);
     });
   }
+  function fetchAllRepoData(url) {
+    return fetch(url)
+      .then(res => res.json())
+      .then(fetchedData => {
+        const navDiv = createAndAppend('div', root, { class: 'nav' });
```

Comment:
The way native `fetch` method works is that if a network response fails, it would still resolve the promise. That's why we need to have an extra check here for a failure scenario. Please refer this documentation - https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch (**Checking that the fetch was successful**)

```
  if(response.ok) {
    return response.json();
  }
  throw new Error('Network response was not ok.');
```